File: ConcatenarPalabras/app/src/main/java/com/example/concatenarpalabras/MainActivity.java
package com.example.concatenarpalabras;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText p1;
    private EditText p2;
    private TextView tv1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        p1 = (EditText) findViewById(R.id.palabra1);
        p2 = (EditText) findViewById(R.id.palabra2);
        tv1 = (TextView) findViewById(R.id.resultado);
    }

    public void concatenar(View view){
        String palabra1 = p1.getText().toString();
        String palabra2 = p2.getText().toString();
        String resultado = palabra1+" "+palabra2;

        tv1.setText(resultado);

    }
}

File: LasBuenasNuevas/app/src/main/java/com/escm1990/lasbuenasnuevas/MainActivity.kt
package com.escm1990.lasbuenasnuevas

import android.graphics.Bitmap
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.webkit.WebChromeClient
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val BASE_URL = "https://blogdebuenasnuevas.blogspot.com/"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //SwipeRefreshLayout para que recargue la pagina con scroll hacia abajo
        idSwipe.setOnRefreshListener {
            idWebView.loadUrl(BASE_URL)
        }

        //creando el cliente webview
        idWebView.webChromeClient = object : WebChromeClient(){ }
        idWebView.webViewClient = object : WebViewClient(){

            //manejar las nuevas cargas de las URL
            override fun shouldOverrideUrlLoading(
                view: WebView?,
                request: WebResourceRequest?
            ): Boolean {
                return false
            }

            //ver cuando una nueva URL se va a cargar
            override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
                super.onPageStarted(view, url, favicon)

                idSwipe.isRefreshing = true
            }

            //que hacer cuando finaliza la carga de la pagina
            override fun onPageFinished(view: WebView?, url: String?) {
                super.onPageFinished(view, url)

                idSwipe.isRefreshing = false
            }

        }

        //activar javascript
        val settings = idWebView.settings
        settings.javaScriptEnabled = true

        //cargar la url
        idWebView.loadUrl(BASE_URL)
    }

    override fun onBackPressed() {
        if(idWebView.canGoBack()){
            idWebView.goBack()
        }else{
            super.onBackPressed()
        }
    }

}